File: Iteration2/src/Domain/RegistrationIDGenerator.java
package Domain;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class RegistrationIDGenerator extends IDGenerator {
    public RegistrationIDGenerator(){
        super();
        this.setDigitCount(5);
        this.setIdPrefix("R");
        this.setIdSuffix("");
        this.setUsedIDs(getUsedIDsByOtherRegistrations());
    }

    private List<String> getUsedIDsByOtherRegistrations(){
        Collection<Registration> allRegistrations = Department.getInstance().getAllRegistrations();
        List<String> alreadyUsedIDs = new ArrayList<>();
       
        for (Registration registration: allRegistrations) {
            alreadyUsedIDs.add(registration.getRegistrationNo());
        }
        return alreadyUsedIDs;
    }
}

File: Iteration3/IDGenerators/RegistrationIDGenerator.py
from IDGenerators.IDGenerator import IDGenerator

class RegistrationIDGenerator(IDGenerator):
    def __init__(self):
        super().__init__()
        self.set_digit_count(5)
        self.set_id_prefix("R")
        self.set_id_suffix("")
        self.set_used_ids([''])  # self.get_used_ids_by_other_registrations()


    def get_used_ids_by_other_registrations(self):
        from Domain.Department import Department
        all_registrations = Department.get_instance().get_all_registrations()
        already_used_ids = [registration.get_registration_no() for registration in all_registrations]
        return already_used_ids

